mysys/crc32/crc32_x86.c

```diff
@@ -508,4 +510,36 @@ crc32_intel_pclmul (u32 *pcrc, const byte *inbuf, size_t inlen)
 #endif
 }
 
+#ifdef __GNUC__
+int crc32_pclmul_enabled(void)
+{
+  int eax, ecx;
+  /* We assume that the CPUID instruction and its parameter 1 are available.
+  We do not support any precursors of the Intel 80486. */
+  asm("cpuid" : "=a"(eax), "=c"(ecx) : "0"(1) : "ebx", "edx");
+  return !(~ecx & (1 << 19 | 1 << 1));
+}
+#elif defined _MSC_VER
+#include <intrin.h>
+int crc32_pclmul_enabled(void)
+{
+  /* We assume that the CPUID instruction and its parameter 1 are available.
+  We do not support any precursors of the Intel 80486. */
+  int regs[4];
+  __cpuid(regs, 1);
+  return !(~regs[2] & (1 << 19 | 1 << 1));
+}
```

Comment:
Until a `pclmul` interface has been implemented for the Microsoft compiler, that check should be written as
```c
#elif 0 /* defined _MSC_VER */ /* FIXME: implement the pclmul interface */
```
Right, @vaintroub ?

Reply:
Windows build passed. If we still need to revamp it let me know.

Reply:
Yes, the build and tests passed on Windows, but last time I looked, the `pclmul` implementation was using the GCC syntax for inline assembler, which was never supported by Microsoft. @vaintroub has previously mentioned that Microsoft prefers compiler built-in functions and does not allow inline assembler for AMD64 at all. I think that he should check what the generated code actually looks like.

Reply:
yes, we copied the non-portable GCC inline assembly from Percona that copied it from elsewhere, but  there are portable, across compilers, Intel intrinsics, that are not only about Microsoft. Check how RocksDB is using pclmul for example. And then you do not need too many #ifdef GCC,  MSC_VER.  Here is pclmul code from Intel itself https://github.com/intel/soft-crc/blob/34a84bfd8278ff48e6aa67f0746618242266c8a2/crcr.h#L195  . If Intel does not use gcc assembly, why should we?

Reply:
So as discussed will file another jira ticket to track this and port intel pclmul.